js/src/utils.ts

```diff
@@ -203,7 +200,10 @@ export async function getAllDomains(
   const accounts = await connection.getProgramAccounts(NAME_PROGRAM_ID, {
     filters,
   });
-  return accounts.map((a) => a.pubkey);
+  const encodedNameArr: PublicKey[] = [];
+  accounts.forEach((a) => encodedNameArr.push(a.pubkey));
+  const names = await reverseLookupBatch(connection, encodedNameArr);
+  return names;
```

Comment:
Have you tested other SDKs (react, vue, cf-worker) that are using this code? I believe they need to be updated as well as part of the PR

Also, it will be better to return not only a string representative but pubkey as well, so users can use both of them if needed. Otherwise, in case pubkey is needed instead of string representative, the user will need to make an additional call to the RPC, which is an absolute overhead since it's done here already.

<hr/>

And a small nit, instead of variable definition + forEach it's better to just do that one:
```ts
const encodedNameArr = accounts.map((a) => a.pubkey)
```
1. It looks simpler to read
2. Now you don't need to define type of the `encodedNameArr`, cause TS automatically casts type based on the map's return type

Reply:
@letehaha Agree that makes sense, as @dr497 mentioned I'll create a separate function that also returns pubkey. Thank you both!

Reply:
@letehaha @dr497 Cleaned up commit history and pushed new changes for your review. In the update I have fixed prettier settings, and separated function that returns human readable domains. Also created a test file for the new function.